lodash.js

```diff
@@ -1978,10 +1978,16 @@
       path = isKey(path, object) ? [path + ''] : toPath(path);
 
       var index = 0,
+          part,
           length = path.length;
 
       while (object != null && index < length) {
-        object = object[path[index++]];
+        part = path[index++];
+        if (isArrayLike(object) && /^-\d+$/.test(part)) {
```

Comment:
The regex tests for both whether the string is number like and negatives. I'm sure there's a better way to do this